realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -59,19 +61,46 @@ private RealmThreadPoolExecutor(int corePoolSize, int maxPoolSize) {
                 new ArrayBlockingQueue<Runnable>(QUEUE_SIZE));
     }
 
-    @Override
-    public Future<?> submit(Runnable task) {
+    /**
+     * Submits a runnable for executing a transaction.
+     *
+     * @param task the task to submit
+     * @return a future representing pending completion of the task
+     */
+    public Future<?> submitTransaction(Realm realm, Runnable task) {
```

Comment:
Yup. We will need it for https://github.com/realm/realm-java/issues/3040 though, and I kinda liked the readability of it. I can revert it until we need it if it is a big problem though?